python/az/aro/azext_aro/_rbac.py

```diff
@@ -37,7 +36,7 @@ def _create_role_assignment(auth_client, resource, params):
             logger.warning("%s; retry %d of %d", ex, retries, max_retries)
 
 
-def assign_contributor_to_vnet(cli_ctx, vnet, object_id):
+def assign_contributor_to_vnet(cli_ctx, vnet, object_id, skip_on_lookup_failure=False):
```

Comment:
how about splitting this into two entirely separate functions assign_contributor_on_vnet and has_contributor_on_vnet (or something).  That pushes the error handling to the caller and should make things more explicit/clearer.  It's more of a refactor, but I think the resulting code will be more maintainable.  Prefer avoiding embedding logic in funky switches.